main.cpp

```diff
@@ -74,23 +74,26 @@ struct T
 
 struct Comparator                                //4
 {
-    T* compare(T* a, T* b) //5
+    T* compare(T* a = nullptr, T* b = nullptr)//5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
-        return nullptr;
+        if (a != nullptr && b != nullptr)
+        {
+        if(a->value < b->value) return a;
+        if(a->value > b->value) return b;
+        }
+        return nullptr;   
     }
 };
 
 struct U
 {
     float f1 { 0 }, f2 { 0 };
-    float multiply(float* newValue)      //12
+    float multiply(float* newValue = nullptr)      //12
     {
         std::cout << "U's f1 value: " << f1 << std::endl;
-        f1 = *newValue;
+        if (newValue != nullptr) f1 = *newValue;
         std::cout << "U's f1 updated value: " << f1 << std::endl;
-        while( std::abs(f2 - f1) > 0.001f )
+        while(std::abs(f2 - f1) > 0.001f)
```

Comment:
While this works, if you don't update `f1`, your `while()` loop never runs.
most other students simply exit the function if `newValue` is nullptr.
please exit early if `newValue == nullptr`
also, the instructions never said to supply a default value.
- [ ] revise.